src/main/java/org/opensearch/neuralsearch/highlight/NeuralHighlighter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.highlight;
+
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+import org.opensearch.search.fetch.subphase.highlight.FieldHighlightContext;
+import org.opensearch.search.fetch.subphase.highlight.HighlightField;
+import org.opensearch.search.fetch.subphase.highlight.Highlighter;
+import org.opensearch.core.common.text.Text;
+
+/**
+ * Neural highlighter that uses ML models to identify relevant text spans for highlighting
+ */
+@Log4j2
+public class NeuralHighlighter implements Highlighter {
+    public static final String NAME = "neural";
+
+    private NeuralHighlighterManager neuralHighlighterManager;
+
+    public void initialize(MLCommonsClientAccessor mlClient) {
+        if (neuralHighlighterManager != null) {
+            throw new IllegalStateException("NeuralHighlighter has already been initialized. Multiple initializations are not permitted.");
+        }
+        this.neuralHighlighterManager = new NeuralHighlighterManager(mlClient);
+    }
+
+    @Override
+    public boolean canHighlight(MappedFieldType fieldType) {
+        return true;
```

Comment:
Make it return to true, which is consistent to other core highlighters. If non-text field query passed in, the highlighter will return null, as a result no highlight part in query response, which is same behavior as other core highlighters.